### PR TITLE
Generate a way to export all Notify.gov user contact information

### PR DESCRIPTION
**Pull request by terrazoon**
## Description

Create a task that runs every 24 hours and generates a list of users with:  user name, user email address, user phone number, what service(s) the user is a member of.  Upload the list to S3 in a known location.

The front end can simply download the report from this known location.

This is the backend portion of https://github.com/GSA/notifications-admin/issues/1218.

## Security Considerations

The information in the report is sensitive.

### CODE REVIEW

**Review by xlorepdarkhelm — APPROVED** [review]
LGTM

**Comment by stvnrlly** [comment]
I'm curious: what's the reasoning for a daily task rather than a by-request generation? I'm guessing that it's about caching the data via S3, but perhaps it's also something else.

If it is about caching, that might not be a concern here. As a tool for platform admin use only, we'd expect this to be run very infrequently. As a result, it might be more reasonable to do it by request and then be able to include newly-created accounts as well.

**Comment by terrazoon** [comment]
> I'm curious: what's the reasoning for a daily task rather than a by-request generation? I'm guessing that it's about caching the data via S3, but perhaps it's also something else.
> 
> If it is about caching, that might not be a concern here. As a tool for platform admin use only, we'd expect this to be run very infrequently. As a result, it might be more reasonable to do it by request and then be able to include newly-created accounts as well.

Well it was about caching, but ok.

File: app/user/rest.py [hunk 1]
```diff
@@ -667,6 +668,12 @@ def update_password(user_id):
     return jsonify(data=user.serialize()), 200
 
 
+@user_blueprint.route("/report-all-users", methods=["GET"])
+def report_all_users():
+    users = dao_report_users()
+    return jsonify(data=users.serialize()), 200
```
Comment (ccostino) [inline app/user/rest.py:674]: I see this is following the same pattern as a bunch of the other endpoints, so I'm assuming the frontend would have to take the JSON and convert it into a CSV or whatever else we'd like to do with it?

**Review by ccostino — COMMENTED** [review]
Thanks, @terrazoon!  Just one quick question on the use of the endpoint itself.

File: app/user/rest.py (hunk 1 above)
Comment (ccostino) [reply app/user/rest.py:674]: We caught up and yes, the frontend will have to account for this endpoint.

**Review by ccostino — APPROVED** [review]
Thanks, @terrazoon!

**Pull request merged**